File: SPOJ/SPOJ-CSUMQ.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	// your code goes here
	int size;
	cin >> size;
	int arr[size];
	cin >> arr[0];
	for(int i = 1; i < size; i++)
	{
		cin >> arr[i];
		arr[i] += arr[i - 1];
	}
	int query;
	cin >> query;
	for(int i = 0; i < query; i++)
	{
		int a, b;
		cin >> a >> b;
		if(a) cout << arr[b] - arr[a - 1] << endl;
		else cout << arr[b] << endl;
	}
	return 0;
}

File: SPOJ/SPOJ-CSUMQ.py
size = int(input())
arr = [int(x) for x in input().split()]
for i in range(1, size):
	arr[i] += arr[i - 1]
query = int(input())
for i in range(query):
	a, b = [int(x) for x in input().split()]
	if a == 0: print(arr[b])
	else: print(arr[b] - arr[a - 1])